Java/src/main/java/com/gildedrose/GildedRose.java

```diff
@@ -1,62 +1,76 @@
 package com.gildedrose;
 
 class GildedRose {
-    Item[] items;
+    Item[] itemArray;
+	
+	public static final String AGED_BRIE = "Aged Brie";
+    public static final String BACKSTAGE_PASSES = "Backstage passes to a TAFKAL80ETC concert";
+    public static final String SULFURAS_HAND_RAGNAROS = "Sulfuras, Hand of Ragnaros";
+    
+    public static final int MIN_QUALITY = 0;
+    public static final int MAX_QUALITY = 50;
 
-    public GildedRose(Item[] items) {
-        this.items = items;
+    public GildedRose(Item[] entryItemArray) {
+        this.itemArray = entryItemArray;
     }
 
     public void updateQuality() {
-        for (int i = 0; i < items.length; i++) {
-            if (!items[i].name.equals("Aged Brie")
-                    && !items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                if (items[i].quality > 0) {
-                    if (!items[i].name.equals("Sulfuras, Hand of Ragnaros")) {
-                        items[i].quality = items[i].quality - 1;
-                    }
-                }
-            } else {
-                if (items[i].quality < 50) {
-                    items[i].quality = items[i].quality + 1;
+        for (int i = 0; i < itemArray.length; i++) {
+            itemArray[i] = updateItemQuality(itemArray[i];
+        }
+    }
+	
+	private Item updateItemQuality(Item entryItem){
+		Item itemUpdated = entryItem;
+        
+        boolean isAgedBrie = itemUpdated.name.equals(AGED_BRIE);
+        boolean isBackStage = itemUpdated.name.equals(BACKSTAGE_PASSES);
+        boolean isSulfuras = itemUpdated.name.equals(SULFURAS_HAND_RAGNAROS);
+		
+		if (!isAgedBrie && !isBackStage && !isSulfuras && itemUpdated.quality > MIN_QUALITY) {
+           itemUpdated.quality -= 1;
+        }		
+		else {
+			if (itemUpdated.quality < MAX_QUALITY) {
+				itemUpdated.quality += 1;
 
-                    if (items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                        if (items[i].sellIn < 11) {
-                            if (items[i].quality < 50) {
-                                items[i].quality = items[i].quality + 1;
-                            }
-                        }
+                 if (isBackstage) {
```

Comment:
O código xa está más limpo pero aínda ten unha complexidade (ciclomática ou cognitiva) moi alta. O seguinte paso sería agrupar estes ifs en funcións. Despois, decidiría que función utilizar según o nome, e quitar aínda máis ifs